model/metadata/metadata.go

```diff
@@ -13,11 +13,34 @@
 
 package metadata
 
-import "github.com/prometheus/common/model"
+import (
+	"strings"
+
+	"github.com/prometheus/common/model"
+)
 
 // Metadata stores a series' metadata information.
 type Metadata struct {
 	Type model.MetricType `json:"type"`
 	Unit string           `json:"unit"`
 	Help string           `json:"help"`
 }
+
+// IsEmpty returns true if metadata structure is empty, including unknown type case.
+func (m Metadata) IsEmpty() bool {
+	return (m.Type == "" || m.Type == model.MetricTypeUnknown) && m.Unit == "" && m.Help == ""
+}
+
+// Equals returns true if m is semantically the same as other metadata.
+func (m Metadata) Equals(other Metadata) bool {
+	if strings.Compare(m.Unit, other.Unit) != 0 || strings.Compare(m.Help, other.Help) != 0 {
+		return false
+	}
+
+	if m.Type != "" && m.Type != model.MetricTypeUnknown {
```

Comment:
This will return true if `m` is unknown and other is a Counter for example. Was that the intention ?
I'd do
```
if m.Type == "" || m.Type == model.MetricTypeUnknown {
  return other.Type == "" || other.Type == model.MetricTypeUnknown
}
return m.Type == other.Type
```